system/lib/wasmfs/file.h

```diff
@@ -240,6 +240,37 @@ class Directory : public File {
   size_t getSize() override { return 4096; }
 };
 
+class Symlink : public File {
+protected:
+  // The target file that this symlink points to. This is constant as symlinks
+  // cannot be modified to point to different things.
+  const std::string target;
+
+  size_t getSize() override;
+
+public:
+  static constexpr FileKind expectedKind = File::SymlinkKind;
+  // Note that symlinks provide a mode of 0 to File. The mode of a symlink does
+  // not matter, so that value will never be read (what matters is the mode of
+  // the target).
+  Symlink(std::string target, backend_t backend)
+    : File(File::SymlinkKind, 0, backend), target(target) {}
+  virtual ~Symlink() = default;
+
+  class Handle : public File::Handle {
+    std::shared_ptr<Symlink> getFile() { return file->cast<Symlink>(); }
+
+  public:
+    Handle(std::shared_ptr<File> symlink) : File::Handle(symlink) {}
+    Handle(Handle&&) = default;
+  };
+
+  Handle locked() { return Handle(shared_from_this()); }
```

Comment:
Do we even need a `Handle` given that Symlinks are entirely immutable?

Reply:
Good point, looks like we don't! Removed.